pkg/reconciler/openshift/tektonaddon/extension.go

```diff
@@ -120,21 +117,25 @@ func (oe openshiftExtension) PostReconcile(ctx context.Context, comp v1alpha1.Te
 			}
 
 			if err := createInstallerSet(ctx, oe.operatorClientSet, addon, manifest, oe.version,
-				miscellaneousResourcesInstallerSet, "addon-openshift"); err != nil {
+				MiscellaneousResourcesInstallerSet, "addon-openshift"); err != nil {
 				return err
 			}
 		}
 		logger.Error("failed to get InstallerSet: %s", err)
 		return err
 	}
 
+	if len(installedTIS.Items) == 0 {
+		return v1alpha1.RECONCILE_AGAIN_ERR
+	}
+
```

Comment:
@sm43 if len is 0 shall we return Reconcile again and wait for sometime or shall we return error itself

Reply:
https://github.com/tektoncd/operator/pull/574/files#r782647681

Reply:
adding a explicit wait in situations like these is a kubernetes anti-pattern.

if the cluster state is not " yet " the desired state, we exit out of the reconcile loop (return RECONCILE_AGAIN) and let the future loops proceed past the check.